source/fuzz/transformation_add_bit_instruction_synonym.cpp

```diff
@@ -0,0 +1,220 @@
+  message_.set_instruction_result_id(instruction_result_id);
+  *message_.mutable_fresh_ids() =
+      google::protobuf::RepeatedField<google::protobuf::uint32>(
+          fresh_ids.begin(), fresh_ids.end());
+}
+
+bool TransformationAddBitInstructionSynonym::IsApplicable(
+    opt::IRContext* ir_context,
+    const TransformationContext& transformation_context) const {
+  auto instruction =
+      ir_context->get_def_use_mgr()->GetDef(message_.instruction_result_id());
+
+  // |instruction| must be defined and must be a bit instruction.
+  if (!instruction || !spvOpcodeIsBit(instruction->opcode())) {
+    return false;
+  }
+
+  // Right now, only integer operands are supported.
+  if (ir_context->get_type_mgr()->GetType(instruction->type_id())->AsVector()) {
+    return false;
+  }
+
+  // TODO: This condition could be relaxed if the index exists as another
+  // integer type. All bit indexes must be defined as 32-bit unsigned integers.
+  uint32_t width = ir_context->get_type_mgr()
+                       ->GetType(instruction->type_id())
+                       ->AsInteger()
+                       ->width();
+  for (uint32_t i = 0; i < width; i++) {
+    if (!fuzzerutil::MaybeGetIntegerConstant(ir_context, transformation_context,
+                                             {i}, 32, false, false)) {
+      return false;
+    }
+  }
+
+  // |message_.fresh_ids.size| must have the exact number of fresh ids required
+  // to apply the transformation.
+  if (static_cast<uint32_t>(message_.fresh_ids().size()) !=
+      GetRequiredFreshIdCount(ir_context, instruction)) {
+    return false;
+  }
+
+  // All ids in |message_.fresh_ids| must be fresh.
+  for (uint32_t fresh_id : message_.fresh_ids()) {
+    if (!fuzzerutil::IsFreshId(ir_context, fresh_id)) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+void TransformationAddBitInstructionSynonym::Apply(
+    opt::IRContext* ir_context,
+    TransformationContext* transformation_context) const {
+  auto bit_instruction =
+      ir_context->get_def_use_mgr()->GetDef(message_.instruction_result_id());
+
+  switch (bit_instruction->opcode()) {
+    case SpvOpBitwiseOr:
+    case SpvOpBitwiseXor:
+    case SpvOpBitwiseAnd:
+      AddBitwiseSynonym(ir_context, transformation_context, bit_instruction);
+      break;
+    default:
+      assert(false && "Should be unreachable.");
+      break;
+  }
+
+  ir_context->InvalidateAnalysesExceptFor(opt::IRContext::kAnalysisNone);
+}
+
+protobufs::Transformation TransformationAddBitInstructionSynonym::ToMessage()
+    const {
+  protobufs::Transformation result;
+  *result.mutable_add_bit_instruction_synonym() = message_;
+  return result;
+}
+
+uint32_t TransformationAddBitInstructionSynonym::GetRequiredFreshIdCount(
+    opt::IRContext* ir_context, opt::Instruction* bit_instruction) {
+  // TODO(https://github.com/KhronosGroup/SPIRV-Tools/issues/3557):
+  // Right now, only certain operations are supported.
```

Comment:
Add one space before "Right" so that it is slightly indented after "TODO" (this means that it shows up in CLion's highlighting as part of the TODO).

Reply:
Done.